src/auto/file_attribute_matcher.rs

```diff
@@ -0,0 +1,70 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib::GString;
+use glib::translate::*;
+use std::fmt;
+
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct FileAttributeMatcher(Shared<gio_sys::GFileAttributeMatcher>);
+
+    match fn {
+        ref => |ptr| gio_sys::g_file_attribute_matcher_ref(ptr),
+        unref => |ptr| gio_sys::g_file_attribute_matcher_unref(ptr),
+        get_type => || gio_sys::g_file_attribute_matcher_get_type(),
+    }
+}
+
+impl FileAttributeMatcher {
+    pub fn new(attributes: &str) -> FileAttributeMatcher {
+        unsafe {
+            from_glib_full(gio_sys::g_file_attribute_matcher_new(attributes.to_glib_none().0))
+        }
+    }
+
+    pub fn enumerate_namespace(&self, ns: &str) -> bool {
+        unsafe {
+            from_glib(gio_sys::g_file_attribute_matcher_enumerate_namespace(self.to_glib_none().0, ns.to_glib_none().0))
+        }
+    }
+
+    pub fn enumerate_next(&self) -> Option<GString> {
```

Comment:
This looks like you'd want to implement an `Iterator` here :) Please add an `iter()` method that returns a custom `Iterator` around this function.

Reply:
Or rather `into_iter()` as the matcher is only really usable once.

Reply:
Yes sir!